shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -162,8 +164,14 @@ - (instancetype)initWithSemanticsObject:(SemanticsObject*)semanticsObject
 
 @end
 
+@interface SemanticsObject ()
+/** Should only be called in conjunction with setting child/parent relationship. */
+- (void)privateSetParent:(SemanticsObject*)parent;
```

Comment:
Yes, the point is we want to control the editing of the parent property.  We want to enforce that editing it happens in conjunction with editing children.  If we were using another language I could make the setter private.  Here i just do it by convention and naming.  We could split SemanticsObject into its own file and make it truly private if you aren't happy with how it's setup.

Reply:
Why can't we just leave it as `_parent = parent`?  Am I just forgetting how Objective C works?

Reply:
Because you are calling it on someone else:
```
void SemanticObject.addChilden(SemanticObject child) {
  _children.append(child);
  child.parent = this;
}
```

We want only SemanticObject to have access to the setter, we don't want the accessibility bridge or whoever fiddling with it (private in a c++ sense).   My hope was that when people see a method called "privateSetParent" they might think twice about who is suppose to be calling it and why.